napari/_vispy/vendored/volume.py

```diff
@@ -375,11 +475,42 @@
         """,
 )
 
+# This is an iso shader for categorical data (e.g., label images)
+ISO_CATEGORICAL_SNIPPETS = dict(
+    before_loop="""
+        vec4 color3 = vec4(0.0);  // final color
+        vec3 dstep = 1.5 / u_shape;  // step to sample derivative
+        gl_FragColor = vec4(0.0);
+    """,
+    in_loop="""
+        // background is assumed to be 0
+        if (val > 0) {
+            // Take the last interval in smaller steps
+            vec3 iloc = loc - step;
+            for (int i=0; i<10; i++) {
+                color = $sample(u_volumetex, iloc);
+                if (color.g > 0) {
```

Comment:
why is this check performed on the g component of the vec4 color? (just trying to understand)

Reply:
The second index is where the sample function returns the texture value (label in this case). To be honest, I'm not sure what is in the other elements of the array. I just copied it from the other frag shaders 😆